Refactor datetimeAlmostEqual

ClovisDj opened the pull request:
This change refactors `datetimeAlmostEqual` method to compare its first `datetime` parameter around a delta `test_settings.MILLISECONDS_THRESHOLD` of its second parameter. 
For now `MILLISECONDS_THRESHOLD=500` but depending on whether we still encounter a false positive with this method, we could tweak this value as we wish.

ClovisDj commented:
> Based off what you said in the PR, this change is expecting dt2 to be within 500ms of dt1, when in reality it is checking if it is within 1000ms. I would either change the ms_threshhold to 250 in the settings or divide in half in the function.

I think it's we have a definition issue here. The length of the interval is `1000ms` however `dt2` is expected to be a the center of this interval meaning it should be within `500ms` from `dt1`.

ajhodges commented:
Since we moved this function to python-common, can you please move this change to that repository? (I know your change isn't merged yet, you can either make a branch off of your existing branch, or add this change to your other PR, or wait until we merge your other PR.)

ClovisDj commented:
> Since we moved this function to python-common, can you please move this change to that repository? (I know your change isn't merged yet, you can either make a branch off of your existing branch, or add this change to your other PR, or wait until we merge your other PR.)

It's already done! I copied this change to the existing `PR` and also added a test case.

ajhodges commented:
Awesome! Thanks :+1:
